Lib/socket.py

```diff
@@ -67,6 +67,11 @@
            "has_dualstack_ipv6", "AddressFamily", "SocketKind"]
 __all__.extend(os._get_exports_list(_socket))
 
+def __getattr__(name):
+    import warnings
+    if name == "SocketType":
+        warnings.warn(f"{name} is deprecated", DeprecationWarning)
```

Comment:
Use `warnings._deprecated` here

Reply:
And I'd put the `import warnings` inside the `if name == "SocketType"` to reduce import time even more.

Reply:
It seems that it needs to specify a remove version. Should I write (3,14)?

Reply:
There's a discussion on DPO concerning how long we should wait before removing a deprecated feature. In general, it's two versions, so the remove version would be (3, 16). But it has been suggested to wait for more versions. Now, it should be *at least* (3, 16).

Reply:
If so, I would keep my current changes, which throw a deprecation when using.

Reply:
We can ask @JelleZijlstra how long we should wait. I think removing it in 3.16 should be fine *if* there aren't a lot of projects that are using it (we don't want to stress them nor do we want to break them) but if there are a lot of projects using them, it's preferrable to postpone it or have no exact removal version.